### PR TITLE
CLOUDP-263619 Local Atlas Experience with AtlasCLI is stable and covered with E2E on Ubuntu, Debian, RHEL

### PR DESCRIPTION
**Pull request by fmenezes**
## Proposed changes
Local Atlas Experience with AtlasCLI is stable and covered with E2E on Ubuntu, Debian, RHEL

_Jira ticket:_ CLOUDP-263619

Closes #[issue number]

## Checklist

- [ ] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added any necessary documentation in document requirements section listed in [CONTRIBUTING.md](https://github.com/mongodb/mongodb-atlas-cli/blob/master/CONTRIBUTING.md) (if appropriate)
- [ ] I have addressed the @mongodb/docs-cloud-team comments (if appropriate)
- [ ] I have updated [test/README.md](https://github.com/mongodb/mongodb-atlas-cli/blob/master/test/README.md) (if an e2e test has been added)
- [ ] I have run `make fmt` and formatted my code

### CODE REVIEW

File: build/ci/evergreen.yml [hunk 1]
```diff
@@ -1758,6 +1758,80 @@ buildvariants:
...
+      - localdev
+    allowed_requesters: ["patch", "ad_hoc", "github_pr"]
+    expansions:
+      <<: *go_linux_version
+    tasks:
+      - name: ".e2e .deployments .local"
+        depends_on:
+        - name: compile
+          variant: "code_health"
+          patch_optional: true
```
Comment (gssbzn) [inline build/ci/evergreen.yml:1818]: delete?
```suggestion
```

Comment (fmenezes) [reply build/ci/evergreen.yml:1818]: @gssbzn I will delete on a follow up PR

Comment (fmenezes) [reply build/ci/evergreen.yml:1818]: done

**Review by gssbzn — APPROVED** [review]
LGTM

File: test/e2e/atlas/deployments_local_nocli_test.go [hunk 2]
```diff
@@ -46,15 +46,29 @@ func TestDeploymentsLocalWithNoCLI(t *testing.T) {
 	req := require.New(t)
 	req.NoError(err)
 
+	bin := "docker"
+	if _, err := exec.LookPath("docker"); err != nil {
```
[the remark this answers is not in the document]
Comment (fmenezes) [reply test/e2e/atlas/deployments_local_nocli_test.go:50]: great catch, fixed everywhere within the cli

**Review by gssbzn — APPROVED** [review]

**Pull request merged**